### ISSUE TITLE
Namespaced selectors and & include the '|' in the result

### ISSUE THREAD

**Issue by nuFaqtz** [postcss/postcss-nested#60]
Version 2.04
When using namespaced selectors like:
```
ui|dock {
    &.explorer {
       float: none;
    }
}
```

the result is:
```
ui|dock|.explorer {
    float: none;
}
```
Notice the extra '|'
In version 2.02 this worked correctly.

**Comment by ai** [comment]
You expect `ui.explorer|dock.explorer`? Do you want to send PR?

**Comment by nuFaqtz** [comment]
Expect `ui|dock.explorer`

**Comment by ai** [comment]
OK. I will look on Friday.

Can you give me some link about `|` in selectors?

**Comment by nuFaqtz** [comment]
Thnx,
[https://developer.mozilla.org/nl/docs/Web/CSS/@namespace](https://developer.mozilla.org/nl/docs/Web/CSS/@namespace)

Again, it did work as expected in 2.0.2

**Comment by ai** [comment]
I added you case to tests and got:

```css
ui|dock.explorer {
  float: none;
}
```

Maybe `|.` came from other tools? Can you create some small project to reproduce the error?

Also try to update `postcss-selector-parser` to latest version.

**Comment by nuFaqtz** [comment]
Thnx for checking it and I ran the test and indeed the result was as you say.
However, I checked my css to see why I created this issue in the first place and found out that the provided test was not complete.
The actual css that fails is
```
ui|dock {
    &.explorer {
        ui|docktab {
            float: none;
        }
    }
}
```
Notice the nested namespaced item and this will result in:
```
ui|dockui|.explorer ui|docktab { 
    float: none; 
}
```

**Comment by ai** [comment]
Standard environment test:

1. Remove all other CSS tools. Left only PostCSS. One-by-one, to find the source.
2. Remove all PostCSS plugins.
3. Then remove all non-CSS code.
4. If error still reproduces on only PostCSS with one plugin and some really small CSS input — create git project and send link here.

**Comment by nuFaqtz** [comment]
Just add the following to index.test.js:

```
it('nested namespaced selectors processed the wrong way, with a copied | character', () => {
    return run(
        'ui|dock { &.explorer { ui|docktab {} } }',
        'ui|dockui|.explorer ui|docktab {}');
});

it('nested namespaced selectors processed the right way', () => {
    return run(
        'ui|dock { &.explorer { ui|docktab {} } }',
        'ui|dockui.explorer ui|docktab {}');
});
```
The first on validates - while it test against the wrong result (with the |)
The second one fails as the correct output is not precessed.

**Comment by ai** [comment]
Thanks.

I will try to fix it today, so it will pass this test (check that it is correct test):

```js
it('supports namespaced selectors', () => {
    return run(
        'ui|dock { &.explorer { ui|docktab {} } }',
        'ui|dock.explorer ui|docktab {}');
});
```

**Comment by nuFaqtz** [comment]
Yep, that's the correct one.
Mucho thanks for taking the time.
Keep up the good work!

JamBo

**Comment by ai** [comment]
The problem is inside selector parser. I opened a issue: https://github.com/postcss/postcss-selector-parser/issues/114

**Comment by ai** [comment]
Issue was fixed in selector parser, but we are waiting for next 3.0 release.

There is blocking issue with Stylelint.

**Comment by ai** [comment]
Selector parser 3.0 was released and I updated dependency https://github.com/postcss/postcss-nested/commit/d76950cd2673093b0497f4dc8f7a3a34abea3c5e

**Comment by ai** [comment]
Released in 3.0
